IM-37 Added a route for adding data to the product table.  Also creat…

johnschroeder opened the pull request:
…ed a database convenience module called impdb.js

johnschroeder committed:
IM-37 Added a route for adding data to the product table.  Also created a database convenience module called impdb.js

NamespaceValentine commented on routes/newProductSubmission.js:
```diff
@@ -0,0 +1,85 @@
+            console.error(err.stack);
+            res.status(503).send("Connection Error: " + err.code);
+        } else {
+            connection.beginTransaction(function(err) {
```
Awesome, glad to see transactions, and using the convenience method is great so as not to clutter the code with standard (not specific to our needs) MySQL code.

NamespaceValentine commented on routes/newProductSubmission.js:
```diff
@@ -0,0 +1,85 @@
+ var express = require("express");
+ var router = express.Router();
+ /* Is there a way to do this without the relative path? */
+ var db = require("../imp_modules/impdb.js");
```
I will add a module that allows us to remove the need for relative path, but for now this works.

NamespaceValentine commented on imp_modules/impdb.js:
```diff
@@ -0,0 +1,27 @@
+/**
```
Yay! our first reusable service ^_^.

NamespaceValentine commented on routes/newProductSubmission.js:
```diff
@@ -0,0 +1,85 @@
+
+ /*
+ Usage:
+ localhost:50001/ProductID/Name/Customer/Description/DateCreated
```
usage is actually: 
localhost:50001/newProductSubmission/ProductID/Name/Customer/Description/DateCreated

I would prefer if we could set a standard for what variables vs static stuff in a route looks like:
localhost:50001/newProductSubmission/:ProductID/:Name/:Customer/:Description/:DateCreated
or 
localhost:50001/newProductSubmission/{ProductID}/{Name}/{Customer}/{Description}/{DateCreated}

or something similar. We should also be defining what each variable means in this comment. For examples take a look at:

http://apidocjs.com/

NamespaceValentine commented:
Other than the usage comment, this all looks good. Will sign off on merge once that is handled, so we can start setting a standard.

johnschroeder commented:
More work to do

johnschroeder committed:
IM-37
 - Changed the name of the folder holding the impdb service to imp_services
 - Removed ProductID from the route so it can autoincrement
 - Fixed usage comment
